File: malai-core/org.malai.ts/src-core/fsm/EpsilonTransition.ts
/*
 * This file is part of Malai.
 * Copyright (c) 2009-2018 Arnaud BLOUIN
 * Malai is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later version.
 * Malai is distributed without any warranty; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.
 */

import {Transition} from "./Transition";
import {OutputState} from "./OutputState";
import {InputState} from "./InputState";

export class EpsilonTransition<E> extends Transition<E> {
    public constructor(srcState: OutputState<E>, tgtState: InputState<E>) {
        super(srcState, tgtState);
    }

    /**
     *
     * @param {*} event
     * @return {boolean}
     */
    accept(event: E): boolean {
        return true;
    }

    /**
     *
     * @param {*} event
     * @return {boolean}
     */
    isGuardOK(event: E): boolean {
        return true;
    }

    /**
     *
     * @return {*[]}
     */
    public getAcceptedEvents(): Set<String> {
        return new Set([]);
    }
}

File: src/main/java/io/interacto/fsm/EpsilonTransition.java
/*
 * Interacto
 * Copyright (C) 2019 Arnaud Blouin
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package io.interacto.fsm;

import java.util.Collections;
import java.util.Set;

public class EpsilonTransition<E> extends Transition<E> {
	public EpsilonTransition(final OutputState<E> srcState, final InputState<E> tgtState) {
		super(srcState, tgtState);
	}

	@Override
	protected boolean accept(final E event) {
		return true;
	}

	@Override
	protected boolean isGuardOK(final E event) {
		return true;
	}

	@Override
	public Set<Object> getAcceptedEvents() {
		return Collections.emptySet();
	}
}
